templates/common/_base/files/configure-ovs-network.yaml

```diff
@@ -159,65 +235,89 @@ contents:
-          continue
-        fi
-        nmcli conn mod uuid ${conn_uuid} connection.master ${new_device}
-      done
+      # Get the new connection uuid
+      new_conn=$(nmcli -g connection.uuid conn show ovs-if-phys0)
+
+      # Setup an exit trap to restore any modifications going further
+      handle_exit_error() {
+        e=$?
+        [ $e -eq 0 ] && exit 0
+        # if there was a problem network isn't coming up, revert for debugging
+        set +e
+        nmcli c show
+        # For some reason RHEL7 requires the interface connection to be brought down explicitly whereas RHCOS does not
+        nmcli conn down ovs-if-phys0
+        nmcli conn up $old_conn
+        exit $e
+      }
+      trap "handle_exit_error" EXIT
 
+      # Update connections with master property set to use the new connection
+      replace_connection_master $old_conn $new_conn
+      replace_connection_master $iface $new_conn
+
+      # For some reason RHEL7 requires the ovs bridge to be brought up explicitly whereas RHCOS does not
+      nmcli conn up br-ex
+      # Bring up the new interface connection
       nmcli conn up ovs-if-phys0
 
       if ! nmcli connection show ovs-if-br-ex &> /dev/null; then
         if nmcli --fields ipv4.method,ipv6.method conn show $old_conn | grep manual; then
           echo "Static IP addressing detected on default gateway connection: ${old_conn}"
           # find and copy the old connection to get the address settings
-          if egrep -l --include=*.nmconnection uuid=$old_conn ${NM_CONN_PATH}/*; then
-            old_conn_file=$(egrep -l --include=*.nmconnection uuid=$old_conn ${NM_CONN_PATH}/*)
+          if egrep -l "^uuid=$old_conn" ${NM_CONN_PATH}/*; then
+            old_conn_file=$(egrep -l "^uuid=$old_conn" ${NM_CONN_PATH}/*)
             cloned=false
           else
             echo "WARN: unable to find NM configuration file for conn: ${old_conn}. Attempting to clone conn"
-            old_conn_file=${NM_CONN_PATH}/${old_conn}-clone.nmconnection
             nmcli conn clone ${old_conn} ${old_conn}-clone
-            cloned=true
-            if [ ! -f "$old_conn_file" ]; then
-              echo "ERROR: unable to locate cloned conn file: ${old_conn_file}"
+            shopt -s nullglob
+            old_conn_files=(${NM_CONN_PATH}/"${old_conn}"-clone*)
+            shopt -u nullglob
+            if [ ${#old_conn_files[@]} -ne 1 ]; then
+              echo "ERROR: unable to locate cloned conn file for ${old_conn}-clone"
               exit 1
             fi
+            old_conn_file="${old_conn_files[0]}"
+            cloned=true
             echo "Successfully cloned conn to ${old_conn_file}"
           fi
           echo "old connection file found at: ${old_conn_file}"
-          new_conn_file=${NM_CONN_PATH}/ovs-if-br-ex.nmconnection
+          old_basename=$(basename "${old_conn_file}" .nmconnection)
+          new_conn_file="${old_conn_file/${NM_CONN_PATH}\/$old_basename/${NM_CONN_PATH}/ovs-if-br-ex}"
```

Comment:
Why is there a backslash here?

Reply:
This is a parameter expansion of the form `${PARAMETER/PATTERN/STRING}`. We want to replace `${NM_CONN_PATH}/$old_basename` for `${NM_CONN_PATH}/ovs-if-br-ex` in `old_conn_file`. We need the extra backslash to escape the forward-slash in the pattern so that it is not interpreted to be the forward-slash in the expression.

Reply:
Got it!  Looks good.